frontend/src/components/Pagination.js

```diff
@@ -6,28 +6,25 @@ import { ButtonGroup, Button } from "react-bootstrap";
 const Pagination = (props) => {
   let renderPaginationBox = (content, pageIndex) => (
     <Button
-      key={`pagination-box-${pageIndex}`}
+      key={`pagination-box-${content}`}
       className="PaginationBox"
-      variant="secondary"
+      variant={(pageIndex === props.currentPageIndex ? "primary" : "secondary")}
       onClick={() =>
         pageIndex !== props.currentPageIndex && props.onSetPage(pageIndex)
       }
     >
-      {pageIndex !== props.currentPageIndex && <a>{content}</a>}
-      {pageIndex === props.currentPageIndex && content}
+    {content === pageIndex ? pageIndex + 1 : content}
     </Button>
   );
 
-  let pagesToRender = props.pageIndexEnd - props.pageIndexStart;
+  let pagesToRender = props.numberOfPages;
   if (!pagesToRender) return <ul />;
   return (
     <ButtonGroup>
       {props.currentPageIndex !== 0 && renderPaginationBox("First", 0)}
```

Comment:
We could probably get away with not having a first and last, as you can do that with the numbers.

Reply:
Yes, but in case there will be dozens of pages, we may only display some of them. For example, we have 100 pages, and the current page is 50, so the pagination shows 46, 47, 48, 49, 50, 51, 52, 53, 54, 55. In this case, first and last button may be useful.

Reply:
Makes a lot of sense!